### PR TITLE
Universal editor doc

### PR DESCRIPTION
**Pull request by irenelagno**
## Description
Documentation for Universal Editor Extensbility 

****
## Related Issue
[SITES-20344](https://jira.corp.adobe.com/browse/SITES-20344) Documentation for Universal Editor Extensibility in UIX Docs

## Motivation and Context

Add documentation about extensible service.  The changes are deployed on dev: https://developer-stage.adobe.com/uix/docs/services/aem-universal-editor/

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -37,6 +37,10 @@ module.exports = {
             title: "AEM Content Fragments Editor",
             path: "/services/aem-cf-editor/"
           },
+          {
+            title: "AEM Universal Editor",
```
Comment (lenaorobei) [inline gatsby-config.js:41]:
```suggestion
            title: "The Universal Editor",
```

File: src/pages/services/aem-universal-editor/api/commons/index.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        rightPanel: {
+            getPanels() {
+            // ..
+            }
+        },
+        canvas: {
+            getRenderers() {
+            // ..
+            }
+           }
```
Comment (lenaorobei) [inline src/pages/services/aem-universal-editor/api/commons/index.md:54]: Needs to be formatted.

File: src/pages/services/aem-universal-editor/api/index.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: AEM Universal Editor Extensibility
+description: Learn how to customize AEM Universal Editor
+contributors:
+  - https://github.com/AdobeDocs/uix
+---
+
+# AEM Universal Editor Extension Points
```
Comment (lenaorobei) [inline src/pages/services/aem-universal-editor/api/index.md:8]:
```suggestion
# The Universal Editor Extension Points
```

File: src/pages/services/aem-universal-editor/api/properties-rails/index.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  rightPanel: {
+    addRails(): ExtensionRail[];
+  };
+};
+
+type ExtensionRail = {
+  id: string;
+  header: string;
+  url: string;
+  icon: DynamicIconVariant;
```
Comment (naydav) [inline src/pages/services/aem-universal-editor/api/properties-rails/index.md:137]: Should be `string`

File: src/pages/services/aem-universal-editor/api/index.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
+title: Universal Editor Extensibility
+description: Learn how to customize Universal Editor
+contributors:
+  - https://github.com/AdobeDocs/uix
+---
+
+# The Universal Editor Extension Points
+
+This section covers the utilization of existing extension points, extension registration, and common methods that can be used in any application that leverages extension points for service customization.
```
Comment (naydav) [inline src/pages/services/aem-universal-editor/api/index.md:10]: Maybe make sense to add link [section](link on extension points index)

File: src/pages/services/aem-universal-editor/api/properties-rails/index.md (hunk 4 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/properties-rails/index.md:137]: fixed

File: src/pages/services/aem-universal-editor/api/item-types-renderers/index.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+export default App
+```
+
+2. Implement custom field renderer
+
+```js
+export default () => {
+    const [isLoaded, setIsLoaded] = useState(true);
```
Comment (kokoc) [inline src/pages/services/aem-universal-editor/api/item-types-renderers/index.md:89]: Not a big issue, but I think it supposed to be `isLoading`

File: src/pages/services/aem-universal-editor/api/properties-rails/index.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+export default () => {
+    const { railId } = useParams();
+    if (!railId) {
+        console.error('Rail id parameter is missed');
+        return;
+    }
+
+    // If you need to interact with an AEM instance
+    // const connection = await attach({ id: extensionId });
```
Comment (kokoc) [inline src/pages/services/aem-universal-editor/api/properties-rails/index.md:107]: Not having this thing will throw the error in the console

File: src/pages/services/aem-universal-editor/api/item-types-renderers/index.md (hunk 6 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/item-types-renderers/index.md]: fixed

File: src/pages/services/aem-universal-editor/api/properties-rails/index.md (hunk 7 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/properties-rails/index.md:107]: fixed

File: src/pages/services/aem-universal-editor/api/commons/index.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Extension registration data must include:
+
+- `id` - string with random extension identifier. This identifier useful for debugging of interaction between Universal
+Editor and extension and needed if extension provides custom UI.
+- `methods` - objects with extension code exposed to the Universal Editor console. All methods are grouped into 
+namespaces that represents extension points provided by the Universal Editor.
+Currently, the following **namespaces** are available:
+- _headerMenu_, that allows to add buttons to the header of the Universal Editor;
+- _rightPanel_, that allows to add custom content under the rails to the right panel of the Universal Editor;
+- _canvas_, that allows to add custom renderer for data types
```
Comment (fe-lix-) [inline src/pages/services/aem-universal-editor/api/commons/index.md:52]: Shouldn't this be another level of bullet points.

File: src/pages/services/aem-universal-editor/api/properties-rails/index.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+ModifierType could be combinations of: `shift+`, `ctrl+`, `alt+`, `cmd+`, `shift+ctrl+`, `shift+alt+`, `shift+cmd+`, `ctrl+alt+`, `ctrl+cmd+`, `alt+cmd+`, `shift+ctrl+alt+`, `shift+ctrl+cmd+`, `shift+alt+cmd+`, `ctrl+alt+cmd+`, `shift+ctrl+alt+cmd+`, ``.
+
+KeyType could be: Alphanumeric keys `(A-Z, 0-9)`, `minus`, `equal`, `bracketleft`, `bracketright`, `backslash`, `semicolon`, `quote`, `comma`, `period`, `slash`, `backquote`, `enter`, `escape`, `backspace`, `tab`, `insert`, `home`, `end`, `pageup`, `pagedown`, `delete`, `arrowup`, `arrowdown`, `arrowleft`, `arrowright`.
+
+Examples: `ctrl+k`, `ctrl+shift+s`, `shift+arrowup`, `c.`
+
+## Limitation
+
+Each click on the icon corresponding to the panel will result in the re-rendering of that panel. The panel content is not cached and will be re-ren
```
Comment (fe-lix-) [inline src/pages/services/aem-universal-editor/api/properties-rails/index.md:162]: re-rendered?

File: src/pages/services/aem-universal-editor/api/events/index.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
+---
+title: Working with the events - Universal Editor Extensibility
+description: Leveraging universal editor events within your application
+contributors:
+  - https://github.com/AdobeDocs/uix
+---
+
+# Working with Events
+
+the Universal Editor sends defined events to remote applications. In case the remote application has no custom event listener for the sent event, a fallback event listener provided by the universal-editor-cors package is executed.
```
Comment (fe-lix-) [inline src/pages/services/aem-universal-editor/api/events/index.md:10]: The?

File: gatsby-config.js [hunk 11]
```diff
@@ -214,7 +218,41 @@ module.exports = {
...
+          {
+            title: "Custom panels in Properties Rail",
+            path: "/services/aem-universal-editor/api/properties-rails/"
+          },
+          {
+            title: "Overriding Default Rendering of Item Types",
+            path: "/services/aem-universal-editor/api/item-types-renderers/"
+          },
+          {
+            title: "Supported events",
```
Comment (fe-lix-) [inline gatsby-config.js:247]: Working with events?
Sending custom events?

File: src/pages/services/aem-universal-editor/api/events/index.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+### Working with Events
+
+The Universal Editor offers a list of events that extensions can subscribe to in order to respond to changes in content or the user interface.
+Refer to this [document](https://experienceleague.adobe.com/en/docs/experience-manager-cloud-service/content/implementing/developing/universal-editor/events) for the available event list.
+
+For instance, here's an example of how to subscribe to the "aue:ui-select" event:
```
Comment (fe-lix-) [inline src/pages/services/aem-universal-editor/api/events/index.md:17]: `aue:ui-select`?

Comment (fe-lix-) [reply src/pages/services/aem-universal-editor/api/events/index.md:17]: Adding the proper quotes.

File: src/pages/services/aem-universal-editor/api/events/index.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+            const guestConnection = await attach({id: extensionId});
+            ...
+            await guestConnection.host.remoteApp.addEventListener('aue:ui-select', console.log('event recieved!'));
+            ...
+        })();
+    }, []);
+```
+
+If your business logic requires sending an event to the Universal Editor, you can use the `dispatchEvent` method.
+Here's an example of how to dispatch the "aue:ui-select" event:
```
Comment (fe-lix-) [inline src/pages/services/aem-universal-editor/api/events/index.md:32]: `aue:ui-select`

**Review by fe-lix- — APPROVED** [review]
A few minimal changes. Looks good.

File: src/pages/services/aem-universal-editor/api/commons/index.md (hunk 8 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/commons/index.md:52]: fixed

File: src/pages/services/aem-universal-editor/api/properties-rails/index.md (hunk 9 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/properties-rails/index.md:162]: fixed

File: src/pages/services/aem-universal-editor/api/events/index.md (hunk 10 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/events/index.md:10]: fixed

File: gatsby-config.js (hunk 11 above)
Comment (irenelagno) [reply gatsby-config.js:247]: Yeah working with events sounds better

File: src/pages/services/aem-universal-editor/api/events/index.md (hunk 13 above)
Comment (irenelagno) [reply src/pages/services/aem-universal-editor/api/events/index.md:32]: fixed

**Pull request merged**